packages/wrangler/src/pages.tsx

```diff
@@ -713,10 +713,40 @@ export const pages: BuilderCallback<unknown, unknown> = (yargs) => {
 
       if (usingFunctions) {
         const scriptPath = join(tmpdir(), "./functionsWorker.js");
+
+        closeSync(openSync(scriptPath, "w"));
+
+        const functionsCompiler = spawn(
+          "node",
+          [
+            "/Users/gregbrimble/Developer/wrangler2/packages/pages-functions-compiler/bin/pages-functions-compiler.mjs",
+            "build",
+            functionsDirectory,
+            "--outfile",
+            scriptPath,
+            "--watch",
+          ],
+          {
+            shell: isWindows(),
+          }
+        );
+
+        functionsCompiler.stdout.on("data", (data) => {
+          console.log(`[pages-functions-compiler]: ${data}`);
+        });
+
+        functionsCompiler.stderr.on("data", (data) => {
+          console.error(`[pages-functions-compiler]: ${data}`);
+        });
+
+        functionsCompiler.on("close", (code) => {
+          console.error(`pages-functions-compiler exited with status ${code}.`);
+        });
+
+        RUNNING_PROCESSES.push(functionsCompiler);
+
         miniflareArgs = {
           scriptPath,
-          buildWatchPaths: [functionsDirectory],
-          buildCommand: `npx @cloudflare/pages-functions-compiler build ${functionsDirectory} --outfile ${scriptPath}`,
```

Comment:
They're not needed any more because the `buildWorker` function now does the watching and building rather than asking Miniflare to handle it. This also means that we're able to easily pick up dependency changes which are outside of the `functionsDirectory`.

And as you point out later, unfortunately we do still need to provide different `miniflareArgs` when there is neither a functions directory nor provided script.